server/src/parse_thread.rs

```diff
@@ -242,7 +242,7 @@ impl ParseThread {
         self.workspace_path.join(path)
     }
     fn updated_lex_or_yacc_file(
-        self: &mut ParseThread,
+        self: &ParseThread,
```

Comment:
Pretty close to giving up on the refactorings that this was intended to realize, seems like this mutable variable isn't the only lifetime stopping us from realizing them. :crying_cat_face: